src/GitHub.VisualStudio/Commands/GoToSolutionOrPullRequestFileCommand.cs

```diff
@@ -198,6 +216,81 @@ void OnBeforeQueryStatus(object sender, EventArgs e)
             Visible = false;
         }
 
+        bool TryNavigateFromHistoryFileQueryStatus(IVsTextView sourceView)
+        {
+            if (teamExplorerContext.Value.ActiveRepository?.LocalPath == null)
+            {
+                // Only available when there's an active repository
+                return false;
+            }
+
+            var filePath = FindPath(sourceView);
+            if (filePath == null)
+            {
+                return false;
+            }
+
+            var objectish = gitHubContextService.Value.FindObjectishForTFSTempFile(filePath);
+            if (objectish == null)
+            {
+                // Not a temporary Team Explorer blob file
+                return false;
+            }
+
+            // Navigate from history file is active
+            Text = "Open File in Solution";
+            Visible = true;
+            return true;
+        }
+
+        bool TryNavigateFromHistoryFile(IVsTextView sourceView)
+        {
+            var repositoryDir = teamExplorerContext.Value.ActiveRepository?.LocalPath;
+            if (repositoryDir == null)
+            {
+                return false;
+            }
+
+            var path = FindPath(sourceView);
+            if (path == null)
+            {
+                return false;
+            }
+
+            var objectish = gitHubContextService.Value.FindObjectishForTFSTempFile(path);
+            if (objectish == null)
+            {
+                return false;
+            }
```

Comment:
The first three if statements from this function and the function above are meant to be the same.
The only minor minor difference _(took me a moment to spot)_ is you are saving the repository dir in the second implementation.

The intent might be clearer if you combine both to one function that returns `repositoryDir` if the conditions are right or `null` if they fail.

Reply:
`GetRepositoryDirectoryForSourceViewAndPath` or something..